File: Year-2023/LeetCode-May-Challenge/MatrixDiagonalSum.cpp
#include<bits/stdc++.h>
using namespace std;
class Solution {
public:
    int diagonalSum(vector<vector<int>>& mat) {
        int res = 0;
        for(int i = 0; i < mat.size(); ++i) res += mat[i][i];
        int j = mat.size() - 1;
        for(int i = 0; i < mat.size(); ++i, j--)
        if(i != j) res += mat[i][j];
        return res;
    }
};
int main(){
    vector<vector<int>> mat = {{1,2,3},
                            {4,5,6},
                            {7,8,9}};
    Solution *s = new Solution();
    cout<<s->diagonalSum(mat)<<endl;
    return 0;
}

File: Year-2023/LeetCode-May-Challenge/MatrixDiagonalSum.js
var diagonalSum = function (mat) {
    let n = mat.length, res = 0;
    for (let i = 0; i < n; ++i) res += mat[i][i];
    let j = n - 1;
    for (let i = 0; i < n; ++i, j--)
        if (i !== j) res += mat[i][j];
    return res;
};

let mat = [[1, 2, 3],
        [4, 5, 6],
        [7, 8, 9]];
console.log(diagonalSum(mat));